dataset/test/data_array_comparison_test.cpp

```diff
@@ -44,17 +44,18 @@ class DataArray_comparison_operators : public ::testing::Test {
     dataset.setCoord(Dim::Y, makeVariable<double>(Dims{Dim::Y}, Shape{3}));
 
     dataset.setCoord(Dim("labels"), makeVariable<int>(Dims{Dim::X}, Shape{4}));
-    dataset.setMask("mask", makeVariable<bool>(Dims{Dim::X}, Shape{4}));
-
-    dataset.setAttr("global_attr", makeVariable<int>(Values{int{}}));
 
     dataset.setData("val_and_var",
                     makeVariable<double>(Dims{Dim::Y, Dim::X}, Shape{3, 4},
                                          Values(12), Variances(12)));
-    dataset.setAttr("val_and_var", "attr", makeVariable<int>(Values{int{}}));
+    dataset.setCoord("val_and_var", Dim("attr"),
```

Comment:
It is, but at the time I chose not to. Having single-argument constructors, especially from simple/common types is often avoided since it adds the risk of converting things that should not. We can probably revisit this decision at some point.